DiscoverySampleApp/DiscoverySampleApp/BrowseView.m

```diff
@@ -0,0 +1,299 @@
+  imgView.image = img;
+  imgView.contentMode = UIViewContentModeScaleAspectFit;
+  self.navigationItem.titleView = imgView;
+
+  // Title label
+  self.videoTitle.text = self.configuration.initasset[@"title"];
+  self.videoTitle.textColor = [UIColor colorWithRed:0.29 green:0.31 blue:0.33 alpha:1.0];
+}
+
+- (void)clearScrollView {
+  dispatch_async(dispatch_get_main_queue(), ^{
+    for (UIView *subview in self.scrollview.subviews) {
+      [subview removeFromSuperview];
+    }
+  });
+}
+
+- (void)viewWillTransitionToSize:(CGSize)size
+       withTransitionCoordinator:(id<UIViewControllerTransitionCoordinator>)coordinator {
+  [coordinator animateAlongsideTransition:^(id<UIViewControllerTransitionCoordinatorContext> context) {
+    UIInterfaceOrientation orientation = UIApplication.sharedApplication.statusBarOrientation;
+    switch (orientation) {
+      case UIDeviceOrientationPortrait:
+      case UIDeviceOrientationPortraitUpsideDown:
+        [self renderPortrait];
+        break;
+
+      default:
+        [self renderLandscape];
+        break;
+    }
+  } completion:^(id<UIViewControllerTransitionCoordinatorContext> context) {
+  }];
+
+  [super viewWillTransitionToSize:size withTransitionCoordinator:coordinator];
+}
+
+- (void)renderPortrait {
+  NSLog(@"User render orientation: portrait");
+
+  [self.navigationController setNavigationBarHidden:NO animated:YES]; //Show navigation controller / logo /
+  self.videoTitle.hidden = NO;
+  self.scrollview.hidden = NO;
+
+  int height = self.view.frame.size.height * percentage; //assignt to a variable to know where the player ends
+  self.playerview.frame = CGRectMake(1, 87, self.view.frame.size.width, height);
+  self.playerViewTop.constant = 80.0;
+  self.playerViewHeight.active = YES;
+  self.playerViewFullHeight.active = NO;
+  self.scrollViewTop.active = YES;
+}
+
+- (void)renderLandscape {
+  NSLog(@"User render orientation: landscape");
+
+  [self.navigationController setNavigationBarHidden:YES animated:YES]; //Hide navigationbar
+  self.videoTitle.hidden = YES;   //remove title uilabel
+  self.scrollview.hidden = YES;
+
+  int height = self.view.frame.size.height * percentage;//assignt to a variable to know where the player ends
+  self.playerview.frame = CGRectMake(1, 1, self.view.frame.size.width, height);
+  self.playerViewTop.constant = 0.0;
+  self.playerViewHeight.active = NO;
+  self.playerViewFullHeight.active = YES;
+  self.scrollViewTop.active = NO;
+}
+
+#pragma mark - PlayerViewDelegate
+
+- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
+  if ([segue.identifier isEqualToString:@"embed"]) {
+    self.playerViewController = segue.destinationViewController;
+    self.playerViewController.delegate = self;
+  }
+}
+
+- (void)didUpdateAssetTitle:(NSString *)title {
+  dispatch_async(dispatch_get_main_queue(), ^{
+    self.videoTitle.text = title;
+  });
+}
+
+- (void)didStartPlaying {
+//  [self updateScrollViewWithDiscovery];
```

Comment:
Should not we remove commented code?

Reply:
No, it's an option that can be also used